language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -134,30 +275,38 @@ computeCek con (Var _ varName)          = do
 -- 2. performs a constant application and calls 'returnCek' ('FrameTyInstArg', 'FrameApplyFun')
 -- 3. puts 'FrameApplyFun' on top of the context and proceeds with the argument from 'FrameApplyArg'
 -- 4. grows the resulting term ('FrameWrap')
-returnCek :: Context -> Plain Value -> CekM EvaluationResultDef
-returnCek []                                  res = pure $ EvaluationSuccess res
-returnCek (FrameTyInstArg ty           : con) fun = instantiateEvaluate con ty fun
+returnCek :: Context -> WithMemory Value -> CekM w s EvaluationResultDef
+returnCek []                                  res = pure $ EvaluationSuccess (void res)
+returnCek (FrameTyInstArg ty           : con) fun = do
+    spendBoth fun 1 1 -- TODO
```

Comment:
Likewise, it's a bit unclear what we should say here. I guess it's still a "step" in the machine, so should have a cost.